app/models/order.rb

```diff
@@ -96,6 +96,11 @@ def can_dispatch_item?
     ORDER_UNPROCESSED_STATES.include?(state)
   end
 
+  def self.counts_for(created_by_id)
+    order_state_counts = group(:state).where(created_by_id: created_by_id).count
```

Comment:
How about  `where(state IN (?), valid_states_array).distinct.group(:state).where(:created_by_id).count` ?

Could please check which is better one and let us know?

Reply:
Sure, will try it out.

Reply:
I benchmarked performance of both queries.  The second query is little slow on an average of 2-3 ms.

Reply:
Added refactored query.